### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by LessGrits**

### CODE REVIEW

File: src/components/AddForm/index.jsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  }, [leaders]);
+
+  const handleChange = e => {
+    setValues({
+      ...values,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const onSubmit = async e => {
```
Comment (LessGrits) [inline src/components/AddForm/index.jsx]: e -> event

File: src/components/Board/index.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      <div className={s.board}>
+        <Controls />
+        <table>
+          <thead>
+            <tr>
+              <th>Position</th>
+              <th>Photo</th>
+              <th>Leader</th>
+              <th>Score</th>
+              {day === historyItems.length && <th>Editing</th>}
```
Comment (LessGrits) [inline src/components/Board/index.jsx]: take out expression from template

File: src/components/AddForm/index.jsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        ...leaderResponse,
+        score: +leaderResponse.score,
+        id: leaderResponse.id,
+      });
+    }
+    handleClose();
+  };
+
+  return (
+    <form onSubmit={e => onSubmit(e)}>
```
Comment (LessGrits) [inline src/components/AddForm/index.jsx]: e -> event

File: src/components/AddForm/index.jsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      <InputLabel htmlFor="user-name">Add user name:</InputLabel>
+      <Input
+        onChange={e => handleChange(e)}
```
Comment (LessGrits) [inline src/components/AddForm/index.jsx]: e -> event

File: src/components/Controls/index.jsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+          backgroundColor: '#4c16d0',
+          fontSize: '13px',
+          textTransform: 'inherit',
+          '&:hover': {
+            backgroundColor: '#7239ff',
+          },
+        }}
+      >
+        Previous day
+      </Button>
```
Comment (LessGrits) [inline src/components/Controls/index.jsx]: take out in a separate btn component,  don`t repeat yourself

File: src/components/Controls/index.jsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+          },
+        }}
+      >
+        Sort by
+      </Button>
+
+      <Button
+        variant="contained"
+        onClick={changeToPreviousDay}
+        disabled={day === 0}
```
Comment (LessGrits) [inline src/components/Controls/index.jsx]: take out expression in isFirstDay()

File: src/components/Controls/index.jsx [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+          },
+        }}
+      >
+        Previous day
+      </Button>
+
+      <Button
+        variant="contained"
+        onClick={changeToNextDay}
+        disabled={day === historyItems.length}
```
Comment (LessGrits) [inline src/components/Controls/index.jsx]: take out expression in js

File: src/components/Controls/index.jsx [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+          },
+        }}
+      >
+        Next day
+      </Button>
+
+      <Button
+        variant="contained"
+        onClick={handleOpen}
+        disabled={day !== historyItems.length}
```
Comment (LessGrits) [inline src/components/Controls/index.jsx]: take out expression in js

File: src/components/Controls/index.jsx [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+        sx={{
+          maxWidth: '140px',
+          backgroundColor: '#8201d0',
+          fontSize: '13px',
+          textTransform: 'inherit',
+          display: day === historyItems.length ? 'block' : 'none',
```
Comment (LessGrits) [inline src/components/Controls/index.jsx]: take out expression in js

File: src/components/Controls/index.jsx [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+      <Button
+        variant="contained"
+        onClick={addNewDayToHistory}
+        disabled={historyItems && getDifference(leadersOfCurrentDay, historyItems.at(-1)?.leaders)}
+        sx={{
+          maxWidth: '140px',
+          backgroundColor: '#8201d0',
+          fontSize: '13px',
+          textTransform: 'inherit',
+          display: day === historyItems.length ? 'block' : 'none',
```
Comment (LessGrits) [inline src/components/Controls/index.jsx]: take out expression in js

File: src/components/Controls/index.jsx [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+          },
+        }}
+      >
+        Add new member
+      </Button>
+
+      <Button
+        variant="contained"
+        onClick={addNewDayToHistory}
+        disabled={historyItems && getDifference(leadersOfCurrentDay, historyItems.at(-1)?.leaders)}
```
Comment (LessGrits) [inline src/components/Controls/index.jsx]: take out expression in js

File: src/components/EditForm/index.jsx [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+  }, [leaders]);
+
+  const handleChange = e => {
+    setValues({
+      ...values,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const onSubmit = async e => {
```
Comment (LessGrits) [inline src/components/EditForm/index.jsx]: event

File: src/components/EditForm/index.jsx [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+  const { leaders, sortType } = useSelector(state => state.leader);
+  const { historyItems, day } = useSelector(state => state.history);
+
+  const editLeaderInList = leaderData => dispatch(editLeader(leaderData, sortType));
+
+  useEffect(() => {
+    dispatch(setLeadersOfCurrentDay(leaders));
+  }, [leaders]);
+
+  const handleChange = e => {
```
Comment (LessGrits) [inline src/components/EditForm/index.jsx]: event

File: src/components/Header/header.module.scss [hunk 14]
```diff
@@ -0,0 +1,57 @@
+.banner {
+  background: #8201d0;
```
Comment (LessGrits) [inline src/components/Header/header.module.scss]: display, width, position, marging, padding at the first, then bg, border-radius and other

File: src/components/Header/header.module.scss [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+  font-weight: 400;
+}
+
+.container__card {
+  position: relative;
+  display: flex;
+  justify-content: space-between;
+  gap: 30px;
+  width: 100%;
+  // max-width: 680px;
```
Comment (LessGrits) [inline src/components/Header/header.module.scss]: remove

File: src/components/History/index.jsx [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+    id: 0,
+    time: new Date().getTime(),
+    leaders: [...leadersOfCurrentDay],
+  };
+
+  const fullHistory = [...historyItems, leadersOfCurrentDayValue];
+  const changeButtonDay = index => dispatch(changeDay(index));
+
+  useEffect(() => {
+    dispatch(changeDay(fullHistory.length - 1));
```
Comment (LessGrits) [inline src/components/History/index.jsx]: take out in const fullHistory.length - 1

File: src/components/History/index.jsx [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+
+  return (
+    <div className={s.wrap}>
+      <p className={s.title}>History</p>
+      <div className={s.btn__container}>
+        {fullHistory.map((_, index) => (
+          <Button
+            key={getHistoryName(index)}
+            variant="contained"
+            disabled={day === index}
```
Comment (LessGrits) [inline src/components/History/index.jsx]: expression

File: src/components/Leader/index.jsx [hunk 18]
```diff
@@ -0,0 +1,86 @@
...
+const Leader = ({ leader = leaderDefaultValue }) => {
+  const [open, setOpen] = useState(false);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+  const position = ordinalSuffixOf(leader.position);
+  const routers = pathRouters();
+  const { historyItems, day } = useSelector(state => state.history);
+  const redirectToLeaderProfile = () => day === historyItems.length && routers.setPath(leader.id);
+
+  const findLeaderPosition = historyItems[0]?.leaders.find(person => person.id === leader.id);
```
Comment (LessGrits) [inline src/components/Leader/index.jsx]: getFirstDayLeaderPosition
const first day = historyItems[0]
const currentLeader = leader 
we have no any peson, we have only leader

File: src/components/Leader/index.jsx [hunk 19]
```diff
@@ -0,0 +1,86 @@
...
+
+  const checkDifferenceOfPosition = person => {
+    const positionDifference = findLeaderPosition && findLeaderPosition.position - person.position;
+    if (positionDifference > 0) {
+      return (
+        <div>
+          <img src="images/green__arrow.png" alt="arrow" />
```
Comment (LessGrits) [inline src/components/Leader/index.jsx]: function that get difference value and return img name

File: src/components/Leader/index.jsx [hunk 20]
```diff
@@ -0,0 +1,86 @@
...
+          <p>+{positionDifference}</p>
```
Comment (LessGrits) [inline src/components/Leader/index.jsx]: function getPositionDifferenceTemplateValue() return neaded value

File: src/components/Leader/index.jsx [hunk 21]
```diff
@@ -0,0 +1,86 @@
...
+      <td className={s.photo}>
+        <img src={getImagePath(leader.picture)} aria-hidden onClick={redirectToLeaderProfile} alt="avatar" />
+      </td>
+      <td className={s.leader}>
+        <span aria-hidden onClick={redirectToLeaderProfile}>
+          {leader.name}
+        </span>
+      </td>
+      <td className={s.score}>{leader.score}</td>
+      {day === historyItems.length && (
```
Comment (LessGrits) [inline src/components/Leader/index.jsx]: expression

File: src/components/Leader/leader.module.scss [hunk 22]
```diff
@@ -0,0 +1,69 @@
...
+      &::after {
+        content: '';
+        width: 100%;
+        display: block;
+        position: absolute;
+        left: 0;
+        bottom: -2px;
+        height: 2px;
+        background-color: rgb(0, 0, 0);
+        transition: width 0.5s;
```
Comment (LessGrits) [inline src/components/Leader/leader.module.scss]: don't repeate yourself

File: src/components/History/history.module.scss [hunk 23]
```diff
@@ -0,0 +1,26 @@
+.wrap {
```
Comment (LessGrits) [inline src/components/History/history.module.scss]: history__wrap

File: src/components/History/history.module.scss [hunk 24]
```diff
@@ -0,0 +1,26 @@
...
+  width: 100%;
+  background-color: rgba(207, 255, 240, 0.45);
+  border-radius: 10px;
+  padding: 10px;
+  min-height: 160px;
+  max-height: 160px;
+  height: 100%;
+  overflow: hidden;
+
+  .title {
```
Comment (LessGrits) [inline src/components/History/history.module.scss]: history__title

File: src/components/Leader/leader.module.scss [hunk 25]
```diff
@@ -0,0 +1,69 @@
...
+      }
+    }
+  }
+}
+
+.photo {
```
Comment (LessGrits) [inline src/components/Leader/leader.module.scss]: leader__photo

File: src/components/LeaderProfile/leaderProfile.module.scss [hunk 26]
```diff
@@ -0,0 +1,35 @@
+.inner {
```
Comment (LessGrits) [inline src/components/LeaderProfile/leaderProfile.module.scss]: bad name

File: src/components/LeaderProfile/leaderProfile.module.scss [hunk 27]
```diff
@@ -0,0 +1,35 @@
...
+  padding: 50px;
+  background: aliceblue;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+}
+
+.wrapper {
```
Comment (LessGrits) [inline src/components/LeaderProfile/leaderProfile.module.scss]: profile__wrap

File: src/components/AddForm/index.jsx [hunk 28]
```diff
@@ -0,0 +1,73 @@
...
+  const [values, setValues] = useState({ name: '', score: 0 });
+  const dispatch = useDispatch();
+  const { leaders, sortType } = useSelector(state => state.leader);
+  const addLeaderInList = leaderData => dispatch(addLeader(leaderData, sortType));
+
+  useEffect(() => {
+    dispatch(setLeadersOfCurrentDay(leaders));
+  }, [leaders]);
+
+  const handleChange = e => {
```
Comment (LessGrits) [inline src/components/AddForm/index.jsx]: event

File: src/components/Board/board.module.scss [hunk 29]
```diff
@@ -0,0 +1,72 @@
...
+        &:last-child {
+          border-bottom: 0;
+        }
+
+        td {
+          font-size: 1.3rem;
+          padding: 5px;
+
+          &:last-child {
+            img {
```
Comment (LessGrits) [inline src/components/Board/board.module.scss:43]: don't understand what is an img element, give him class name

File: src/components/Board/board.module.scss [hunk 30]
```diff
@@ -0,0 +1,72 @@
...
+              max-width: 20px;
+              width: 100%;
+            }
+          }
+
+          @media (max-width: 500px) {
+            font-size: 18px;
+          }
+
+          svg {
```
Comment (LessGrits) [inline src/components/Board/board.module.scss]: don't understand what is an svg element, give him class name

File: src/components/Board/board.module.scss [hunk 31]
```diff
@@ -0,0 +1,72 @@
...
+            cursor: pointer;
+            opacity: 0.5;
+          }
+        }
+      }
+    }
+  }
+}
+
+.container {
```
Comment (LessGrits) [inline src/components/Board/board.module.scss]: what is container? board-container?

File: src/components/Board/board.module.scss [hunk 32]
```diff
@@ -0,0 +1,72 @@
...
+  position: relative;
+}
+
+.wrap__progress {
```
Comment (LessGrits) [inline src/components/Board/board.module.scss]: progress-wrap

File: src/components/Controls/controls.module.scss [hunk 33]
```diff
@@ -0,0 +1,28 @@
+.wrapControl {
```
Comment (LessGrits) [inline src/components/Controls/controls.module.scss]: controlWrap

File: src/components/Header/header.module.scss [hunk 34]
```diff
@@ -0,0 +1,57 @@
...
+
+.title {
+  font-family: 'Roboto', sans-serif;
+  font-size: 2.5rem;
+  text-align: center;
+  padding: 1rem 0;
+  font-weight: 400;
+}
+
+.container__card {
```
Comment (LessGrits) [inline src/components/Header/header.module.scss]: card__container

File: src/components/Header/header.module.scss [hunk 35]
```diff
@@ -0,0 +1,57 @@
...
+  min-height: 160px;
+  z-index: 2;
+
+  @media (max-width: 980px) {
+    flex-wrap: wrap;
+    justify-content: space-around;
+  }
+}
+
+.wrap__progress {
```
Comment (LessGrits) [inline src/components/Header/header.module.scss]: progress__wrap

File: src/redux/leader/reducer.js [hunk 36]
```diff
@@ -0,0 +1,57 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { orderTypes } from 'helpers/consts';
+import { reverseOrderType, sortListByOrder } from 'helpers/functions';
+
+const initialState = {
+  leaders: [],
```
Comment (LessGrits) [inline src/redux/leader/reducer.js]: leaderboardList

File: src/redux/store.js [hunk 37]
```diff
@@ -0,0 +1,13 @@
+import { configureStore } from '@reduxjs/toolkit';
+import historyReducer from 'redux/history/reducer';
+import leaderReducer from 'redux/leader/reducer';
+
+const store = configureStore({
+  reducer: {
+    leader: leaderReducer,
```
Comment (LessGrits) [inline src/redux/store.js]: leaderboard

File: src/components/Leader/leader.module.scss [hunk 38]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+.position {
+  font-family: 'Roboto', sans-serif;
+  font-weight: 600;
+  color: rgb(220, 220, 220);
+}
+
+.diffrence {
+  div {
```
Comment (LessGrits) [inline src/components/Leader/leader.module.scss]: bad practise use default tag name for  selectors

File: src/components/Controls/index.jsx [hunk 39]
```diff
@@ -0,0 +1,122 @@
...
+
+  const { historyItems, day, leadersOfCurrentDay } = useSelector(state => state.history);
+  const sortedLeaders = () => dispatch(sortLeaders());
+  const changeToPreviousDay = () => dispatch(changeDay(day - 1));
+  const changeToNextDay = () => dispatch(changeDay(day + 1));
+  const addNewDayToHistory = () => dispatch(addNewDay(leadersOfCurrentDay));
+  const getDifference = (array1, array2) => JSON.stringify(array1) === JSON.stringify(array2);
+
+  return (
+    <div className={s.wrapControl}>
```
Comment (LessGrits) [inline src/components/Controls/index.jsx]: control__wrap

File: src/components/Leader/leader.module.scss [hunk 40]
```diff
@@ -0,0 +1,69 @@
...
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    gap: 10px;
+  }
+}
+
+.leader {
+  & > span {
```
Comment (LessGrits) [inline src/components/Leader/leader.module.scss]: bad practise

File: src/components/Leader/leader.module.scss [hunk 41]
```diff
@@ -0,0 +1,69 @@
...
+      }
+    }
+  }
+}
+
+.photo {
+  display: flex;
+  justify-content: center;
+
+  & > img {
```
Comment (LessGrits) [inline src/components/Leader/leader.module.scss]: bad practices

File: src/components/LeaderProfile/leaderProfile.module.scss [hunk 42]
```diff
@@ -0,0 +1,35 @@
...
+    align-items: center;
+    gap: 5rem;
+  }
+
+  @media (min-width: 1024px) {
+    grid-template-columns: minmax(300px, 350px) 1fr;
+  }
+}
+
+.container {
```
Comment (LessGrits) [inline src/components/LeaderProfile/leaderProfile.module.scss]: profile__container

File: src/components/LeaderProfile/leaderProfile.module.scss [hunk 43]
```diff
@@ -0,0 +1,35 @@
...
+  height: 100%;
+}
+
+.title {
```
Comment (LessGrits) [inline src/components/LeaderProfile/leaderProfile.module.scss]: bad name

File: src/redux/leader/actions.js [hunk 44]
```diff
@@ -0,0 +1,17 @@
...
+import { startLoading, leadersFetchingError, setLeaders } from 'redux/leader/reducer';
+
+export const fetchLeaders = () => async dispatch => {
+  try {
+    dispatch(startLoading());
+    const response = await axios.get(`${API.LEADERS}`);
+    const preparedData = setZeroScore(response.data);
+    dispatch(setLeaders(preparedData));
+    dispatch(setLeadersOfCurrentDay(preparedData));
+  } catch (e) {
```
Comment (LessGrits) [inline src/redux/leader/actions.js]: error

File: src/shared/components/CardLeaderTop/cardLeaderTop.module.scss [hunk 45]
```diff
@@ -0,0 +1,24 @@
...
+  }
+
+  p {
+    text-align: center;
+  }
+
+  h3 {
+    text-align: center;
+    padding: 5px 0px;
+  }
```
Comment (LessGrits) [inline src/shared/components/CardLeaderTop/cardLeaderTop.module.scss]: use class name

File: src/shared/components/Wrapper/wrapper.module.scss [hunk 46]
```diff
@@ -0,0 +1,6 @@
+.inner {
```
Comment (LessGrits) [inline src/shared/components/Wrapper/wrapper.module.scss]: wrraper__content

File: src/App.jsx [hunk 47]
```diff
@@ -0,0 +1,24 @@
...
+import LeaderProfile from 'components/LeaderProfile';
+import NotFound from 'components/NotFound';
+import 'App.css';
+
+const App = () => {
+  return (
+    <div className="App">
+      <Wrapper>
+        <Routes>
+          <Route path="/" element={<Board />} />
```
Comment (funkynine) [inline src/App.jsx:14]: U can put here element={Board}

File: src/App.jsx [hunk 48]
```diff
@@ -0,0 +1,24 @@
...
+          <Route path="leader">
```
Comment (funkynine) [inline src/App.jsx]: Maybe we can write leader:id 
Check please

File: src/components/AddForm/index.jsx [hunk 49]
```diff
@@ -0,0 +1,73 @@
...
+
+import { Input, InputLabel, Button } from '@mui/material';
+import { createLeader } from 'shared/api/requests/leaders';
+import { addLeader } from 'redux/leader/reducer';
+import { setLeadersOfCurrentDay } from 'redux/history/reducer';
+
+const AddForm = ({ handleClose }) => {
+  const [values, setValues] = useState({ name: '', score: 0 });
+  const dispatch = useDispatch();
+  const { leaders, sortType } = useSelector(state => state.leader);
```
Comment (funkynine) [inline src/components/AddForm/index.jsx]: create selectors in redux 
example: 
useSelector(getLeaders);

File: src/components/AddForm/index.jsx [hunk 50]
```diff
@@ -0,0 +1,73 @@
...
+      ...values,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const onSubmit = async e => {
+    e.preventDefault();
+    const leaderResponse = await createLeader(values);
+    if (leaderResponse) {
+      addLeaderInList({
```
Comment (funkynine) [inline src/components/AddForm/index.jsx]: create const payload = {
        ...leaderResponse,
        score: +leaderResponse.score,
        id: leaderResponse.id,
      }. and put to addLeaderInList(payload)

File: src/components/AddForm/index.jsx (hunk 1 above)
Comment (funkynine) [inline src/components/AddForm/index.jsx]: never use the short name for params 
e -> event

File: src/components/AddForm/index.jsx (hunk 3 above)
Comment (funkynine) [inline src/components/AddForm/index.jsx]: add formik to project and create from with formik

File: src/components/Board/index.jsx [hunk 51]
```diff
@@ -0,0 +1,58 @@
...
+
+import { Header, Controls, Leader } from 'components';
+import { fetchLeaders } from 'redux/leader/actions';
+import { getHistory } from 'redux/history/actions';
+import s from './board.module.scss';
+
+const Board = () => {
+  const dispatch = useDispatch();
+  const { leaders, isLoading } = useSelector(state => state.leader);
+  const { historyItems, day } = useSelector(state => state.history);
```
Comment (funkynine) [inline src/components/Board/index.jsx]: selector func

File: src/components/Controls/controls.module.scss [hunk 52]
```diff
@@ -0,0 +1,28 @@
...
+
+  @media (max-width: 1100px) {
+    gap: 12px;
+  }
+  @media (max-width: 425px) {
+    flex-direction: column;
+  }
+}
+
+button {
```
Comment (funkynine) [inline src/components/Controls/controls.module.scss]: global style 
move this styles to global scss file

File: src/components/Controls/controls.module.scss (hunk 33 above)
Comment (funkynine) [inline src/components/Controls/controls.module.scss]: why here camelCase?

File: src/components/Controls/index.jsx [hunk 53]
```diff
@@ -0,0 +1,122 @@
+import { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
```
Comment (funkynine) [inline src/components/Controls/index.jsx:3]: Please use sort for imports in your files

File: src/components/EditForm/index.jsx [hunk 54]
```diff
@@ -0,0 +1,79 @@
...
+import { Input, InputLabel, Button } from '@mui/material';
+import { updateLeader } from 'shared/api/requests/leaders';
+import { setLeadersOfCurrentDay } from 'redux/history/reducer';
+
+const leaderDefaultValue = { name: '', score: 0, position: 0, id: '', picture: '' };
+
+const EditForm = ({ leader = leaderDefaultValue, handleClose }) => {
+  const [values, setValues] = useState(leader);
+  const dispatch = useDispatch();
+  const { leaders, sortType } = useSelector(state => state.leader);
```
Comment (funkynine) [inline src/components/EditForm/index.jsx]: selector func

File: src/components/Header/header.module.scss (hunk 15 above)
Comment (funkynine) [inline src/components/Header/header.module.scss]: delete

File: src/components/History/index.jsx [hunk 55]
```diff
@@ -0,0 +1,67 @@
...
+import { setLeaders } from 'redux/leader/reducer';
+import { Button } from '@mui/material';
+import { ordinalSuffixOf } from 'helpers/functions';
+import s from './history.module.scss';
+
+const History = () => {
+  const dispatch = useDispatch();
+  const { historyItems, day, leadersOfCurrentDay } = useSelector(state => state.history);
+
+  const leadersOfCurrentDayValue = {
```
Comment (funkynine) [inline src/components/History/index.jsx]: use here useMemo

File: src/components/Leader/index.jsx [hunk 56]
```diff
@@ -0,0 +1,86 @@
...
+import { pathRouters } from 'utility/pathRouters';
+import { useSelector } from 'react-redux';
+import EditForm from 'components/EditForm';
+import s from './leader.module.scss';
+
+const leaderDefaultValue = { name: '', score: 0, position: 0, id: '', picture: '' };
+
+const Leader = ({ leader = leaderDefaultValue }) => {
+  const [open, setOpen] = useState(false);
+  const handleOpen = () => setOpen(true);
```
Comment (funkynine) [inline src/components/Leader/index.jsx:20]: What happens here is functions, hooks .....

File: src/components/Leader/index.jsx (hunk 18 above)
Comment (funkynine) [inline src/components/Leader/index.jsx]: useMemo

File: src/components/Leader/index.jsx [hunk 57]
```diff
@@ -0,0 +1,86 @@
...
+  const handleClose = () => setOpen(false);
+  const position = ordinalSuffixOf(leader.position);
+  const routers = pathRouters();
+  const { historyItems, day } = useSelector(state => state.history);
+  const redirectToLeaderProfile = () => day === historyItems.length && routers.setPath(leader.id);
+
+  const findLeaderPosition = historyItems[0]?.leaders.find(person => person.id === leader.id);
+
+  const checkDifferenceOfPosition = person => {
```
Comment (funkynine) [inline src/components/Leader/index.jsx]: Create component personPosition

File: src/components/Leader/leader.module.scss [hunk 58]
```diff
@@ -0,0 +1,69 @@
...
+  color: #d27a00;
+}
+
+.position {
+  font-family: 'Roboto', sans-serif;
+  font-weight: 600;
+  color: rgb(220, 220, 220);
+}
+
+.diffrence {
```
Comment (funkynine) [inline src/components/Leader/leader.module.scss]: read doc about BEM

**Pull request merged**